src/parser/SparqlParserHelpers.h

```diff
@@ -28,7 +28,7 @@ struct ResultOfParseAndRemainingText {
 
 struct ParserAndVisitor {
  private:
-  string input_;
+  string_view input_;
   antlr4::ANTLRInputStream stream_{input_};
```

Comment:
This is dangerous concerning the life time. I think the `ParserAndvisitor` should store a copy of the input as a string. I am sorry for the confusion. You can revert back to `std::string` and `move` where you don't need the string anymore. I am sorry for the trouble.